python/ffsim/hamiltonians/molecular_hamiltonian.py

```diff
@@ -156,6 +158,81 @@ def _fermion_operator_(self) -> FermionOperator:
                 }
             )
         return op
+    
+
+    @staticmethod
+    def from_fermion_operator(op: FermionOperator) -> MolecularHamiltonian:
+        """Convert a FermionOperator to a MolecularHamiltonian."""
+        # extract number of spatial orbitals
+        norb = 1 + max(orb for term in op for _, _, orb in term)
+
+        # initialize constant, one‑ and two‑body tensors
+        constant: float = 0.0
+        one_body_tensor = np.zeros((norb, norb), dtype=complex)
+        two_body_tensor = np.zeros((norb, norb, norb, norb), dtype=complex)
+
+        # track which (p,q,r,s) we've already processed
+        seen_2b: Dict[Tuple[int,int,int,int], None] = {}
+
+        for term, coeff in op.items():
+            # constant term: empty tuple
+            if not term:
+                constant = coeff.real
```

Comment:
Actually, we should check that `coeff.imag` is zero, and if not, raise an error. Can you add a test to catch this? I realize now we didn't do that for DiagonalCoulombHamiltonian.